--- vmware-server-distrib/bin/vmware-config.pl~	2008-04-07 21:04:12.095539024 +0300
+++ vmware-server-distrib/bin/vmware-config.pl	2008-04-07 21:08:37.755436186 +0300
@@ -3932,6 +3932,26 @@
       && can_build_module($name) eq 'yes') {
     db_add_answer('BUILDR_' . $name, 'yes');
   } else {
+
+    my $kmod_dir = $cKernelModuleDir . '/'. $gSystem{'uts_release'};
+    if (-d $kmod_dir) {
+      if (not open(MODULESDEP, $kmod_dir .'/modules.dep')) {
+         error("Unable to open kernel module dependency file\n.");
+      }
+
+      my $found;
+      my $match = qr/\Q$name\E\.ko(?:\.gz)?:$/;
+      while (<MODULESDEP>) {
+         if ($match) {
+            $found = 'yes';
+            last;
+         }
+      }
+      close(MODULESDEP);
+
+      return $found if $found;
+    }
+
     my @mod_list;
 
     @mod_list = get_suitable_modules($mod_dir . '/binary');
