--- vmware-server-distrib/bin/vmware-config.pl~	2008-04-08 08:31:09.000000000 +0300
+++ vmware-server-distrib/bin/vmware-config.pl	2008-04-08 09:17:56.468815748 +0300
@@ -7671,10 +7671,6 @@
          . '>/dev/null 2>&1');
 
   configure_superserver();
-  if (vmware_product() ne 'server') {
-    # authd pam file is automatically installed in ESX Server
-    configure_wgs_pam_d();
-  }
   fix_vmlist_permissions();
 }
 
